### PR TITLE
SNOW-698565 SnowPark interface cannot be analysed by mypy (missing typed.py)

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-698565 and #621 where using mypy as linter tool would not work.

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [x] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Add py.typed file and package with setup.py, modify tox.ini to include mypy dependency for a simple integration test.

### CODE REVIEW

**Comment by sfc-gh-lspiegelberg** [comment]
I have read the CLA Document and I hereby sign the CLA

File: tests/integ/test_mypy.py
Comment (sfc-gh-lspiegelberg) [reply tests/integ/test_mypy.py:13]: should we put this into a new ticket and merge this PR first as one to enable running mypy at all? We have a good amount of typing issues right now, which would require resolution before making it a merge gate.

File: tox.ini
Comment (sfc-gh-lspiegelberg) [reply tox.ini]: just checked again, it's actually `bash` that needs to be added here (just pushed a fix). This solves warnings when running tox, that will become an error once tox is upgraded to tox4.

File: tests/unit/test_mypy.py
Comment (sfc-gh-lspiegelberg) [reply tests/unit/test_mypy.py:31]: You can reproduce them via `mypy -p snowflake.snowpark --explicit-package-bases`, but it's a long list of typing issues

```
src/snowflake/snowpark/row.py:76: error: Signature of "__getitem__" incompatible with supertype "tuple"  [override]
src/snowflake/snowpark/row.py:76: note:      Superclass:
src/snowflake/snowpark/row.py:76: note:          @overload
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, SupportsIndex, /) -> Any
src/snowflake/snowpark/row.py:76: note:          @overload
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, slice, /) -> Tuple[Any, ...]
src/snowflake/snowpark/row.py:76: note:      Subclass:
src/snowflake/snowpark/row.py:76: note:          def __getitem__(self, Union[int, str, slice], /) -> Any
src/snowflake/snowpark/row.py:211: error: Incompatible return value type (got "Union[Dict[Any, Any], Iterable[Dict[Any, Any]]]", expected "Dict[Any, Any]")  [return-value]
src/snowflake/snowpark/row.py:224: error: List comprehension has incompatible type List[Union[Dict[Any, Any], Iterable[Dict[Any, Any]]]]; expected List[Dict[Any, Any]]  [misc]
src/snowflake/snowpark/row.py:243: error: Incompatible return value type (got "Optional[Any]", expected "bool")  [return-value]
src/snowflake/snowpark/_internal/code_generation.py:35: error: List or tuple literal expected as the defaults argument to namedtuple()  [misc]
src/snowflake/snowpark/_internal/code_generation.py:150: error: Argument 1 to "get_func_references" has incompatible type "object"; expected "FunctionType"  [arg-type]
src/snowflake/snowpark/_internal/code_generation.py:180: error: Invalid index type "Optional[int]" for "Tuple[str, ...]"; expected type "SupportsIndex"  [index]
src/snowflake/snowpark/_internal/code_generation.py:379: error: Argument 2 to "range" has incompatible type "Optional[int]"; expected "SupportsIndex"  [arg-type]
src/snowflake/snowpark/_internal/code_generation.py:383: error: Unsupported operand types for - ("None" and "int")  [operator]
src/snowflake/snowpark/_internal/code_generation.py:383: note: Left operand is of type "Optional[int]"
src/snowflake/snowpark/_internal/code_generation.py:439: error: Missing positional argument "alias" in call to "ImportNameAliasPair"  [call-arg]
src/snowflake/snowpark/_internal/code_generation.py:475: error: Missing positional argument "alias" in call to "ImportNameAliasPair"  [call-arg]
src/snowflake/snowpark/_internal/error_message.py:272: error: Argument 1 to "SnowparkSQLException" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/_internal/error_message.py:278: error: Argument 1 to "SnowparkSQLException" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/types.py:328: error: Name "pandas.Series" is not defined  [name-defined]
src/snowflake/snowpark/types.py:335: error: "TypeVarTuple" support is experimental, use --enable-incomplete-feature=TypeVarTuple to enable  [misc]
src/snowflake/snowpark/types.py:337: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/types.py:337: error: Free type variable expected in Generic[...]  [misc]
src/snowflake/snowpark/context.py:19: error: Incompatible return value type (got "Optional[Session]", expected "Session")  [return-value]
src/snowflake/snowpark/_internal/utils.py:286: error: The return type of a generator function should be "Generator" or one of its supertypes  [misc]
src/snowflake/snowpark/_internal/utils.py:448: error: Incompatible return value type (got "Optional[Dict[str, str]]", expected "Dict[str, str]")  [return-value]
src/snowflake/snowpark/file_operation.py:192: error: Incompatible types in assignment (expression has type "str", target has type "int")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:211: error: Missing return statement  [return]
src/snowflake/snowpark/_internal/type_utils.py:240: error: Incompatible types in assignment (expression has type "Type[DataType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:249: error: Incompatible types in assignment (expression has type "Type[DataType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:254: error: Incompatible types in assignment (expression has type "Type[StringType]", target has type "Type[_FractionalType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:295: error: Incompatible types in assignment (expression has type "dict_items[Any, Any]", variable has type "zip[Tuple[Any, None]]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:334: error: "DataType" has no attribute "fields"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:355: error: "DataType" has no attribute "element_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:361: error: "DataType" has no attribute "key_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:362: error: "DataType" has no attribute "value_type"  [attr-defined]
src/snowflake/snowpark/_internal/type_utils.py:477: error: Argument 1 to "parse_arg_annotation" has incompatible type "slice"; expected "expr"  [arg-type]
src/snowflake/snowpark/_internal/type_utils.py:543: error: Need type annotation for "DATA_TYPE_STRING_OBJECT_MAPPINGS" (hint: "DATA_TYPE_STRING_OBJECT_MAPPINGS: Dict[<type>, <type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/type_utils.py:547: error: Incompatible types in assignment (expression has type "Type[ShortType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:548: error: Incompatible types in assignment (expression has type "Type[ByteType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:549: error: Incompatible types in assignment (expression has type "Type[LongType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:550: error: Incompatible types in assignment (expression has type "Type[DecimalType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:551: error: Incompatible types in assignment (expression has type "Type[DecimalType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:552: error: Incompatible types in assignment (expression has type "Type[MapType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:553: error: Incompatible types in assignment (expression has type "Type[ArrayType]", target has type "Type[IntegerType]")  [assignment]
src/snowflake/snowpark/_internal/type_utils.py:561: error: Missing return statement  [return]
src/snowflake/snowpark/_internal/type_utils.py:580: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:581: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:584: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, str]")  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:587: error: Argument 2 to NewType(...) must be subclassable (got VALID_PYTHON_TYPES_FOR_LITERAL_VALUE?)  [valid-newtype]
src/snowflake/snowpark/_internal/type_utils.py:587: error: Variable "snowflake.snowpark._internal.type_utils.VALID_PYTHON_TYPES_FOR_LITERAL_VALUE" is not valid as a type  [valid-type]
src/snowflake/snowpark/_internal/type_utils.py:587: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
src/snowflake/snowpark/_internal/type_utils.py:588: error: Argument 2 to NewType(...) must be subclassable (got "Union[Column, LiteralType]")  [valid-newtype]
src/snowflake/snowpark/_internal/telemetry.py:239: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "TelemetryClient")  [assignment]
src/snowflake/snowpark/_internal/telemetry.py:308: error: Incompatible types in assignment (expression has type "List[str]", target has type "str")  [assignment]
src/snowflake/snowpark/_internal/telemetry.py:310: error: Incompatible types in assignment (expression has type "List[str]", target has type "str")  [assignment]
src/snowflake/snowpark/_internal/udf_utils.py:24: error: Skipping analyzing "cloudpickle": module is installed, but missing library stubs or py.typed marker  [import]
src/snowflake/snowpark/_internal/udf_utils.py:24: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
src/snowflake/snowpark/_internal/udf_utils.py:79: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:90: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:92: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:97: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:98: error: Value of type "Union[Callable[..., Any], Tuple[str, str]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:129: error: Incompatible return value type (got "Tuple[Optional[DataType], List[DataType]]", expected "Tuple[DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:199: error: Value of type "Optional[List[DataType]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:200: error: Argument 1 to "len" has incompatible type "Optional[List[DataType]]"; expected "Sized"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:201: error: Value of type "Optional[List[DataType]]" is not indexable  [index]
src/snowflake/snowpark/_internal/udf_utils.py:205: error: List comprehension has incompatible type List[Union[DataType, None, Any]]; expected List[DataType]  [misc]
src/snowflake/snowpark/_internal/udf_utils.py:206: error: Item "None" of "Optional[Iterable[DataType]]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:209: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:213: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:217: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:232: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:237: error: Item "Tuple[str, ...]" of "Union[Callable[..., Any], Tuple[str, str]]" has no attribute "__code__"  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:239: error: Item "Tuple[str, ...]" of "Union[Callable[..., Any], Tuple[str, str]]" has no attribute "__code__"  [union-attr]
src/snowflake/snowpark/_internal/udf_utils.py:250: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:254: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:259: error: List comprehension has incompatible type List[Optional[DataType]]; expected List[DataType]  [misc]
src/snowflake/snowpark/_internal/udf_utils.py:264: error: Incompatible return value type (got "Tuple[bool, bool, Optional[DataType], List[DataType]]", expected "Tuple[bool, bool, DataType, List[DataType]]")  [return-value]
src/snowflake/snowpark/_internal/udf_utils.py:268: error: "DataType" has no attribute "element_type"  [attr-defined]
src/snowflake/snowpark/_internal/udf_utils.py:385: error: Argument 1 to "generate_source_code" has incompatible type "Callable[..., Any]"; expected "Union[FunctionType, BuiltinFunctionType]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:554: error: Argument 2 to "isinstance" has incompatible type "_SpecialForm"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/udf_utils.py:560: error: Argument 1 to "generate_python_code" has incompatible type "Union[Callable[..., Any], Tuple[str, str]]"; expected "Callable[..., Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:27: error: Need type annotation for "COLUMN_DEPENDENCY_EMPTY"  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:31: error: Need type annotation for "result" (hint: "result: Set[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:36: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:39: error: Argument 1 to "update" of "set" has incompatible type "Optional[Set[str]]"; expected "Iterable[Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:60: error: Incompatible return value type (got "FrozenSet[Any]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:84: error: Incompatible types in assignment (expression has type "None", variable has type "str")  [assignment]
src/snowflake/snowpark/_internal/analyzer/expression.py:97: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:169: error: Incompatible types in assignment (expression has type "Optional[Set[str]]", variable has type "FrozenSet[str]")  [assignment]
src/snowflake/snowpark/_internal/analyzer/expression.py:186: error: Incompatible return value type (got "FrozenSet[str]", expected "Optional[Set[str]]")  [return-value]
src/snowflake/snowpark/_internal/analyzer/expression.py:194: error: Argument 2 to "isinstance" has incompatible type "Tuple[object, ...]"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:196: error: Argument 1 to "PLAN_CANNOT_CREATE_LITERAL" of "SnowparkClientExceptionMessages" has incompatible type "Type[Any]"; expected "str"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/expression.py:283: error: Item "None" of "Optional[List[Expression]]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/analyzer/expression.py:287: error: List or tuple expected as variadic arguments  [misc]
src/snowflake/snowpark/_internal/analyzer/expression.py:312: error: Need type annotation for "exps" (hint: "exps: List[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/expression.py:337: error: List or tuple expected as variadic arguments  [misc]
src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py:129: error: Argument 1 to "hexlify" has incompatible type "Union[List[Any], bytes, bytearray]"; expected "Union[bytes, Union[bytearray, memoryview, array[Any], mmap, _CData, PickleBuffer]]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:67: error: Function "builtins.any" is not valid as a type  [valid-type]
src/snowflake/snowpark/stored_procedure.py:67: note: Perhaps you meant "typing.Any" instead of "any"?
src/snowflake/snowpark/stored_procedure.py:84: error: Item "None" of "Optional[Session]" has no attribute "_conn"  [union-attr]
src/snowflake/snowpark/stored_procedure.py:422: error: Argument 2 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:423: error: Argument 3 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[List[DataType]]"; expected "List[DataType]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:424: error: Argument 4 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Union[str, Iterable[str], None]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:530: error: Argument 2 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:531: error: Argument 3 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Optional[List[DataType]]"; expected "List[DataType]"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:532: error: Argument 4 to "_do_register_sp" of "StoredProcedureRegistration" has incompatible type "Union[str, Iterable[str], None]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:637: error: Argument 3 to "cleanup_failed_permanent_registration" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/snowflake/snowpark/stored_procedure.py:641: error: Argument 1 to "StoredProcedure" has incompatible type "Union[Callable[..., Any], Tuple[str, str]]"; expected "Callable[..., Any]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/unary_expression.py:33: error: Argument 1 to "derive_dependent_columns" has incompatible type "Optional[Expression]"; expected "Expression"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:31: error: Incompatible types in assignment (expression has type "Type[NullsFirst]", base class "SortDirection" defined the type as "NullOrdering")  [assignment]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:36: error: Incompatible types in assignment (expression has type "Type[NullsLast]", base class "SortDirection" defined the type as "NullOrdering")  [assignment]
src/snowflake/snowpark/_internal/analyzer/sort_expression.py:55: error: Argument 1 to "derive_dependent_columns" has incompatible type "Optional[Expression]"; expected "Expression"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py:16: error: Need type annotation for "children" (hint: "children: List[<type>] = ...")  [var-annotated]
src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py:84: error: Cannot override writeable attribute with read-only property  [override]
src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py:176: error: Cannot override writeable attribute with read-only property  [override]
src/snowflake/snowpark/window.py:32: error: Incompatible types in assignment (expression has type "UnboundedPreceding", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:34: error: Incompatible types in assignment (expression has type "Literal", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:39: error: Incompatible types in assignment (expression has type "UnboundedFollowing", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/window.py:41: error: Incompatible types in assignment (expression has type "Literal", variable has type "CurrentRow")  [assignment]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:410: error: Incompatible types in assignment (expression has type "float", variable has type "int")  [assignment]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:445: error: Argument 2 to "to_sql" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:647: error: Argument 1 to "convert_sp_to_sf_type" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:1149: error: Argument 1 to "convert_sp_to_sf_type" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py:1156: error: Argument 1 to "schema_expression" has incompatible type "Optional[DataType]"; expected "DataType"  [arg-type]
src/snowflake/snowpark/column.py:81: error: Incompatible return value type (got "Union[ColumnOrLiteral, DataFrame]", expected "Column")  [return-value]
src/snowflake/snowpark/column.py:82: error: Argument 2 to "isinstance" has incompatible type "Tuple[object, ...]"; expected "Union[type, Tuple[Union[type, Tuple[Any, ...]], ...]]"  [arg-type]
src/snowflake/snowpark/column.py:218: error: Incompatible types in assignment (expression has type "UnresolvedAttribute", variable has type "Star")  [assignment]
src/snowflake/snowpark/column.py:220: error: Incompatible types in assignment (expression has type "Expression", variable has type "Star")  [assignment]
src/snowflake/snowpark/column.py:233: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]
src/snowflake/snowpark/column.py:233: note: This violates the Liskov substitution principle
src/snowflake/snowpark/column.py:233: note: (Skipping most remaining errors due to unresolved imports or missing stubs; fix these first)
src/snowflake/snowpark/async_job.py:289: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:399: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:428: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:574: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/_internal/server_connection.py:575: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/session.py:17: error: Skipping analyzing "cloudpickle": module is installed, but missing library stubs or py.typed marker  [import]
src/snowflake/snowpark/session.py:1167: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/session.py:1313: error: Name "pandas.DataFrame" is not defined  [name-defined]
src/snowflake/snowpark/dataframe_reader.py:234: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
src/snowflake/snowpark/dataframe.py:145: error: Cannot find implementation or library stub for module named "table"  [import]
src/snowflake/snowpark/dataframe.py:667: error: Name "to_pandas" already defined on line 657  [no-redef]
src/snowflake/snowpark/dataframe.py:735: error: Name "to_pandas_batches" already defined on line 725  [no-redef]
Found 134 errors in 25 files (checked 51 source files)
```

Comment (sfc-gh-sfan) [reply tests/unit/test_mypy.py:31]: I see. I guess fixing the typing issue is going to be a separate PR?

Comment (sfc-gh-jdu) [reply tests/unit/test_mypy.py:31]: Yea let's create a jira for that?

**Review by sfc-gh-jdu — APPROVED** [review]
@sfc-gh-mkeller please take a look

File: setup.py
Comment (sfc-gh-achandrasekaran) [reply setup.py:79]: Yes, adding new dependencies needs legal signoff. @sfc-gh-aling did it for another library and can help. You need to send an email to legal explaining the library and the license

File: tox.ini
Comment (sfc-gh-mkeller) [inline tox.ini:23]: Why is bash used here? We shouldn't need this

File: tests/unit/test_mypy.py
Comment (sfc-gh-mkeller) [inline tests/unit/test_mypy.py:8]: This isn't the right way to enforce `mypy` in the code-base.
Please add it to our `.pre-commit-config,yaml` instead, see https://github.com/pre-commit/mirrors-mypy

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

File: tests/unit/test_mypy.py
Comment (sfc-gh-mkeller) [reply tests/unit/test_mypy.py:8]: You'll want to add the other dependencies to `additional_dependencies`

File: setup.py
Comment (sfc-gh-stan) [reply setup.py:79]: https://developer-handbook.c1.us-west-2.aws-dev.app.snowflake.com/docs/reference/security-review/open-source-software-review/

Comment (sfc-gh-lspiegelberg) [reply setup.py:79]: I've reached out to the legal team, let's see what they say. I agree with @sfc-gh-mkeller that mypy as a merge gate and precommit would be a good idea. Would suggest to spin out a new issue for this, including adding a precommit and fixing existing typing issues.

File: tox.ini
Comment (sfc-gh-lspiegelberg) [reply tox.ini:23]: It disables a warning when running tox.

File: tests/unit/test_mypy.py
Comment (sfc-gh-lspiegelberg) [reply tests/unit/test_mypy.py:8]: will be handled in a separate issue https://snowflakecomputing.atlassian.net/browse/SNOW-705882

File: setup.py
Comment (sfc-gh-mkeller) [reply setup.py:79]: > Would suggest to spin out a new issue for this, including adding a precommit and fixing existing typing issues.

I'm not convinced that running `mypy` in a test (that just fails anyway) brings any value to the project. You're adding dependencies that might as well not be there. We might as well not do anything else other than add `py.typed` and add it to `package_data`.

Comment (sfc-gh-mkeller) [reply setup.py:79]: > [developer-handbook.c1.us-west-2.aws-dev.app.snowflake.com/docs/reference/security-review/open-source-software-review](https://developer-handbook.c1.us-west-2.aws-dev.app.snowflake.com/docs/reference/security-review/open-source-software-review/)

`mypy` falls under exceptions by that site I think

File: tests/unit/test_mypy.py
Comment (sfc-gh-lspiegelberg) [reply tests/unit/test_mypy.py]: removed test to address concerns, mypy integration deferred to other issue.

**Review by sfc-gh-mkeller — APPROVED** [review]
Address my 1 comment before merging, otherwise

:ship: :ship:

File: tox.ini
Comment (sfc-gh-lspiegelberg) [reply tox.ini:23]: After removing this the CI fails, so I readded it. Let's remove this line in your PR together with the script

File: tox.ini
Comment (sfc-gh-lspiegelberg) [inline tox.ini:83]: required as well for CI to work.

**Pull request merged**
